integration-tests/cli/discount-codes.it.js

```diff
@@ -277,4 +247,116 @@ describe('DiscountCode tests', () => {
+              typeId: 'cart-discount',
+              id: cartDiscount.id,
+            },
+          ],
+        })
+      ))
+      return createData(apiConfig, 'discountCodes', preparedDiscountCodes)
+      .catch(process.stderr.write)
+    })
+
+    test('should write json output to file by default', (done) => {
+      const jsonFilePath = tmp.fileSync().name
+      const expected = {
+        version: 1,
+        name: {
+          en: 'Sammy',
+          de: 'Valerian',
+        },
+        description: {
+          en: 'greatest promo',
+          de: 'super angebot',
+        },
+        cartDiscounts: [{
+          typeId: 'cart-discount',
+          id: cartDiscount.id,
+        }],
+        cartPredicate: 'lineItemTotal(1 = 1) >  "10.00 USD"',
+        isActive: true,
+        maxApplications: 10,
+        maxApplicationsPerCustomer: 2,
+      }
+
+      exec(`${bin} -o ${jsonFilePath} -p ${projectKey}`,
+        (cliError, stdout, stderr) => {
+          expect(cliError).toBeFalsy()
+          expect(stderr).toBeFalsy()
+          expect(stdout).toMatch(/Operation completed successfully/)
+
+          fs.readFile(jsonFilePath, { encoding: 'utf8' }, (error, data) => {
+            expect(error).toBeFalsy()
+            const actual = JSON.parse(data)
+            actual.forEach((codeObj) => {
+              expect(codeObj).toMatchObject(expected)
+              expect(isuuid(codeObj.id)).toBe(true)
+              expect(codeObj.createdAt).toMatch(UTCDateTimeRegex)
+              expect(codeObj.lastModifiedAt).toMatch(UTCDateTimeRegex)
+            })
+
+            done()
+          })
+        },
+      )
+    })
+
+    test('should write csv output to file when passed the option', (done) => {
+      const csvFilePath = tmp.fileSync().name
+      const expected = {
+        version: '1',
+        name: {
+          en: 'Sammy',
+          de: 'Valerian',
+        },
+        description: {
+          en: 'greatest promo',
+          de: 'super angebot',
+        },
+        cartDiscounts: cartDiscount.id,
+        cartPredicate: 'lineItemTotal(1 = 1) >  "10.00 USD"',
+        isActive: 'true',
+        maxApplications: '10',
+        maxApplicationsPerCustomer: '2',
+      }
+
+      exec(`${bin} -o ${csvFilePath} -p ${projectKey} -f csv`,
+        (cliError, stdout, stderr) => {
+          expect(cliError).toBeFalsy()
+          expect(stderr).toBeFalsy()
+          expect(stdout).toMatch(/Operation completed successfully/)
+
+          fs.readFile(csvFilePath, { encoding: 'utf8' }, (error, data) => {
+            expect(error).toBeFalsy()
+            csv().fromString(data)
+              .on('json', (jsonObj) => {
+                expect(jsonObj).toMatchObject(expected)
+                expect(isuuid(jsonObj.id)).toBe(true)
+                expect(jsonObj.createdAt).toMatch(UTCDateTimeRegex)
+                expect(jsonObj.lastModifiedAt).toMatch(UTCDateTimeRegex)
+              })
+              .on('done', () => done())
```

Comment:
This will make failing tests pass, as the handler would simply call done on errors and the assertions would just be ignored, passing the tests :-)